### PR TITLE
Update xnack option and move setting of AMDGPU_TARGET before find_package(hip)

### PR DESCRIPTION
**Pull request by neon60**
* Update xnack option and added gfx803
* Update gitlab CI
* Move setting of AMDGPU_TARGET before find_package(hip)

### CODE REVIEW

**Comment by saadrahim** [comment]
@doctorcolinsmith Are we stopping to build for gfx803?

**Review by doctorcolinsmith — APPROVED** [review]
Still need to check if gfx803 can come off the list, but we will deal with that later.

File: CMakeLists.txt [hunk 1]
```diff
@@ -68,8 +68,12 @@ include(cmake/Dependencies.cmake)
 set(VERSION_STRING "2.10.7")
 rocm_setup_version(VERSION ${VERSION_STRING})
 
-# AMD targets
-set(AMDGPU_TARGETS gfx900:xnack-;gfx906:xnack-;gfx908:xnack- CACHE STRING "List of specific machine types for library to target")
+# Use target ID syntax if supported for AMDGPU_TARGETS
+if(TARGET_ID_SUPPORT)
```
[the remark this answers is not in the document]
Comment (neon60) [reply CMakeLists.txt:81]: Good catch. I missed that part.

**Review by stanleytsang-amd — APPROVED** [review]

**Comment by neon60** [comment]
The CI passed. Can we merge in?

**Comment by stanleytsang-amd** [comment]
> The CI passed. Can we merge in?

Sorry.  Done.

**Pull request merged**